app/serializers/timetable_serializer.rb

```diff
@@ -1,11 +1,12 @@
-class TimetableSerializer < ActiveModel::Serializer
-  type 'timetable_entries'
+# frozen_string_literal: true
 
-  attributes :start_time, :reason, :nomis_type
+class TimetableSerializer
+  include JSONAPI::Serializer
 
-  belongs_to :location, serializer: LocationSerializer
+  set_type :timetable_entries
 
-  def nomis_type
-    object.type
-  end
+  attributes :start_time, :reason
+  attribute :nomis_type, &:type
+
+  belongs_to :location, &:location
```

Comment:
Not sure why we need to pass the proc with the same key name?

Reply:
This one is fun - the timetable model doesn't use a traditional `belongs_to :location` relationship, but instead has a custom `location` method. The serializer likes to use `location_id` by default to resolve associations which isn't implemented on the timetable model. So this forces use of the custom `location` method instead - alternatively I could add an implementation for `location_id` on the timetable model. There's another Nomis model (court cases maybe) that has the same pattern.

Reply:
Wow 🤯

Reply:
Okay, well in which case this is a clean solution